src/nupic/types/Sdr.hpp

```diff
@@ -506,6 +508,23 @@ class SparseDistributedRepresentation : public Serializable
      */
     void load(std::istream &inStream) override;
 
+		CerealAdapter;
+    template<class Archive>
+    void save_ar(Archive & ar) const
+    {
+        getSparse(); // to make sure sparse is valid.
+        ar(cereal::make_nvp("SDR", dimensions_), cereal::make_nvp("sparse", sparse_) );
```

Comment:
"SDR" - dimensions_, should be "dimensions"-dimensions_;

Reply:
is this i JSON serialization then? Or the NVP will be ignored for binary stream?

Reply:
> "SDR" - dimensions_, should be "dimensions"-dimensions_;

We cannot nest these make_nvp( ) calls and I was looking for a way to include the SDR tag.  And since Cereal nor JSON care what name you give it I thought this would be a good solution.  But you are correct, that is the dimensions_ field.  I will change it to `dimensions`.

> is this a JSON serialization then? Or the NVP will be ignored for binary stream?

This is ignored in the two binary streams.  You can also use them in the load_ar( ) but they are ignored by all streams.